ui/src/shared/components/DashboardRoute.tsx

```diff
@@ -1,31 +1,88 @@
 import React, {PureComponent} from 'react'
+import qs from 'qs'
 import {connect} from 'react-redux'
 import {withRouter, WithRouterProps} from 'react-router'
 import {setDashboard} from 'src/shared/actions/currentDashboard'
-import {AppState} from 'src/types'
+import {getVariables} from 'src/variables/selectors'
+import {selectValue} from 'src/variables/actions/creators'
+import {AppState, Variable} from 'src/types'
 
 interface StateProps {
+  variables: Variable[]
   dashboard: string
 }
 
 interface DispatchProps {
   updateDashboard: typeof setDashboard
+  selectValue: typeof selectValue
 }
 
 type Props = StateProps & DispatchProps & WithRouterProps
 
 class DashboardRoute extends PureComponent<Props> {
-  check(props) {
-    const {dashboard, updateDashboard} = props
+  pendingVars: [{[key: string]: any}]
+
+  syncVariables(props, urlVars) {
     const dashboardID = props.params.dashboardID
+    const {variables, selectValue} = props
+
+    variables.forEach(v => {
```

Comment:
this would really benefit from some commenting. when names are so close, it's easy for reader to get confused